janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/HasStepFolder.java

```diff
@@ -227,6 +226,33 @@ public OrderEntry(String key, Order order) {
             this.key = key;
             this.order = order;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            OrderEntry that = (OrderEntry) o;
+
+            if (key != null ? !key.equals(that.key) : that.key != null) return false;
```

Comment:
This is very interesting; I hadn't seen a `?:` **inside** an `if` condition before. A bit complex to read, methinks.

I believe this is equivalent to the following but with fewer constructs per line/clause so should be easier to understand:

```java
if (key == null && that.key != null) return false;
if (!key.equals(that.key)) return false;
```

Thoughts?

Reply:
Ah yes, a ternary operator nested in an if! I use Intellij to generate `equals` and `hashCode` methods and I agree, it is a bit baroque. I'll update it per your suggestion for improved readability.

Reply:
Hmm, thinking about it some more, that'll cause a NPE if `key == null` and `that.key == null` are both true. This needs some more thought; looks like the original one is well thought-through and correct but perhaps a bit complex.

May be best to leave it alone; if it's the IntelliJ auto-generated method, it's probably well-designed and well-written.